async_blp/handler_refdata.py

```diff
@@ -0,0 +1,54 @@
+import asyncio
+from typing import List
+
+try:
+    import blpapi
+except:
+    from tests import env_test as blpapi
+
+
+class HandlerRef:
+    connection = False
+    service_name = "//blp/refdata"
+    request_name = "ReferenceDataRequest"
+
+    def __init__(self):
+        self.loop = asyncio.get_running_loop()
+        self.event = asyncio.Event()
+        session_options = blpapi.SessionOptions()
+        session_options.setServerHost("localhost")
+        session_options.setServerPort(8194)
+        self.__result = []
+
+        self.session = blpapi.Session(options=session_options,
+                                      eventHandler=self)
+        self.session.startAsync()
+
+    def send_requests(self, requests: List):
+        self.requests = requests
+        self.event.clear()
```

Comment:
Probably this should be done using `call_soon_threadsafe`

Reply:
it's can be called from diff thread